### PR TITLE
Fix sql function error with sources

### PR DESCRIPTION
**Pull request by ivanmalagon**
Related to https://github.com/CartoDB/support/issues/1322
----

This PR fixes a bug in SQL function analysis.

The problem is that the analysis set a `source` attribute but the analysis reference has `primary_source`. It only failed when drag & dropping the source node of the analysis to create another layer. The code has been modified to use the correct key `primary_source`.

There is code to make the existing analyses work.

### Acceptance

Please create SQL analysis **before** deploying so you can assert that this PR doesn't break existing analyses.

### CODE REVIEW

**Comment by elenatorro** [comment]
What does it mean *There is code to make the existing analyses work* ? :)

**Comment by ivanmalagon** [comment]
This function
https://github.com/CartoDB/cartodb/pull/13732/files#diff-65cb91753866426a1ecfded0286fd8a3R52

first try to get the `source` attribute and if it's undefined it goes for `primary_source`. If we go only for `primary_source` in the case of SQL function analysis we would break existing analyses that rely on `source` attribute. If there were no existing SQL function analyses we could build a more elegant solution getting `source` or `primary_source` or whatever depending on the analysis reference. But we can't 😞

File: lib/assets/javascripts/builder/components/modals/add-analysis/analysis-option-models/deprecated-sql-function-option-model.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+   */
+  getFormAttrs: function (layerDefModel) {
+    var attrs = AnalysisOptionModel.prototype.getFormAttrs.apply(this, arguments);
+    delete attrs.source;
+
+    _.extend(attrs, {
+      primary_source: layerDefModel.get('source')
+    });
+
+    return attrs;
```
[the remark this answers is not in the document]
Comment (ivanmalagon) [reply lib/assets/javascripts/builder/components/modals/add-analysis/analysis-option-models/deprecated-sql-function-option-model.js]: It follows the same pattern as the rest of `xxx-options-model.js`

File: lib/assets/test/spec/builder/components/modals/add-analysis/analysis-option-models/deprecated-sql-function-option-model.spec.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        source: 'z0'
+      });
+
+      var formAttrs = model.getFormAttrs(fakeLayerDefModel);
+
+      expect(AnalysisOptionModel.prototype.getFormAttrs).toHaveBeenCalledWith(fakeLayerDefModel);
+      expect(formAttrs).toEqual({ primary_source: 'z0' });
+    });
+  });
+});
```
[the remark this answers is not in the document]
Comment (ivanmalagon) [reply lib/assets/test/spec/builder/components/modals/add-analysis/analysis-option-models/deprecated-sql-function-option-model.spec.js]: It gets tested as stated in https://github.com/CartoDB/cartodb/pull/13732/files/05e1afced3611bbff9ad6d90836db8a3218c5443#diff-8165f8da6d3076592b4a7f5fe3a4d02bR10

When we check that the returned object is `primary_source: 'z0'` we are checking that there is no `source: 'y0'`

File: lib/assets/test/spec/builder/editor/layers/layer-content-view/analyses/analyses-form-models/base-analysis-form-model.spec.js [hunk 3]
```diff
@@ -164,6 +164,54 @@ describe('editor/layers/layer-content-views/analyses/analysis-form-models/base-a
...
+      this.model.set('primary_source', 'z1', { silent: true });
+    });
+
+    it('should return first the `source` node definition model', function () {
+      spyOn(this.model._layerDefinitionModel, 'findAnalysisDefinitionNodeModel').and.callFake(function (name) {
+        return name === 'a0'
+          ? 'the a0'
+          : name === 'z1'
+            ? 'the z1'
+            : null;
```
[the remark this answers is not in the document]
Comment (ivanmalagon) [reply lib/assets/test/spec/builder/editor/layers/layer-content-view/analyses/analyses-form-models/base-analysis-form-model.spec.js]: You're right. It's not readable.

Changing it.

File: lib/assets/javascripts/builder/components/modals/add-analysis/analysis-option-models/deprecated-sql-function-option-model.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivanmalagon) [reply lib/assets/javascripts/builder/components/modals/add-analysis/analysis-option-models/deprecated-sql-function-option-model.js]: I mean, I agree with your point but I'll let it as in the rest of similar modules.

**Pull request merged**